File: WEEK1_C/MARIO-MORE/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height = get_int("How tall do you want the pyramids to be? ");
    while (height <= 0 || height > 8)
    {
        // PROMPT FOR HEIGHT UNTIL A VALID NUMBER IS ENTERED
        height = get_int("Please type a number between 0 and 8 ");
    }

    // PRINT THE PYRAMID
    for (int i = 0; i < height; i++)
    {
        for (int j = height; j > i + 1; j--)
        {
            printf(" ");
        }

        // PRINT THE LEFT SIDE
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        // PRINT THE GAP
        printf("  ");

        // PRINT THE RIGHT SIDE
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: WEEK6_PYTHON/SENTIMENTAL_MARIO-MORE/mario.py
from cs50 import get_int

# PROMPT THE USER FOR THE HEIGHT OF THE PYRAMID
height = get_int('Height: ')

# ENSURE THAT THE HEIGHT IS BETWEEN 1 ND 8
while (height < 1 or height > 8):
    height = get_int('Height: ')

for i in range(height, 0, -1):
    # PRINT SPACES BEFORE THE LEFT PYRAMID
    for j in range(height, height - i + 1, -1):
        print(' ', end='')

    # PRINT THE RIGHT PYRAMID
    for j in range(height - i + 1):
        print('#', end='')

    # PRINT THE GAP IN BETWEEN
    print('  ', end='')

    # PRINT THE LEFT PYRAMID
    for j in range(height - i + 1):
        print('#', end='')

    print()  # PRINT A NEW LINE
